### PR TITLE
Re-enable commented test and make it pass on travis ...

### LINKED ISSUE #157: Monitor proxy_test.go
In proxy_test.go, there are two tests: TestProxy and TestReadyProxy. 
ReadyProxy only tests for the Ready functionality, but this is already tested in TestProxy. The issue in Travis VM is that running both tests sequentially fails, but running one after the other does not. It may be a closing connection - related issue, not sure. It works fine on my Mac anyway so that's strange.
For the moment, I just commented out TestReadyProxy.

### PR DESCRIPTION
**Pull request by Liamsi**
Should fix #157

### CODE REVIEW

**Comment by Liamsi** [comment]
We really should mock `net.Connection` instead of using real connections. Otherwise in every test we open ports, we also test if the port is open etc.  

Done. Please review.

**Comment by Liamsi** [comment]
Golang has a [default mocking framework](https://github.com/golang/mock/tree/master/sample) btw

File: lib/monitor/measure.go [hunk 1]
```diff
@@ -107,6 +111,7 @@ func send(v interface{}) {
 	if !enabled {
 		return
 	}
+	// XXX comment needed:
```
Comment (ineiti) [inline lib/monitor/measure.go]: For a large number of clients (˜10'000), the connection phase can take some time. This is a linear backoff to enable connection even when there are a lot of requests.

File: lib/monitor/proxy_test.go [hunk 2]
```diff
@@ -80,69 +80,67 @@ func TestProxy(t *testing.T) {
...
+	done := make(chan bool)
+	go func() {
+		if err := monitor.Listen(); err != nil {
+			dbg.Error("Could not start listener:", err)
+			dbg.Error("Retry once in 500ms ...")
+			time.Sleep(500 * time.Millisecond)
+			if err := monitor.Listen(); err != nil {
+				dbg.Fatal("Could not start listener:", err)
+			}
+		}
```
Comment (ineiti) [inline lib/monitor/proxy_test.go]: Does it work without this? Will say, is that a remnant of a try to make it work, or is it needed? 
If needed {why? Is it not possible to make TestProxy wait for correct closing of the ports?} else { remove }

File: lib/monitor/proxy_test.go [hunk 3]
```diff
@@ -80,69 +80,67 @@ func TestProxy(t *testing.T) {
...
+	proxyAddr := "localhost:" + strconv.Itoa(SinkPort)
+	if err := ConnectSink(proxyAddr); err != nil {
+		dbg.Error("Could not connect to proxy:", err)
+		dbg.Error("Retry once in 500ms ...")
+		time.Sleep(500 * time.Millisecond)
+		if err := ConnectSink(proxyAddr); err != nil {
+			t.Error(fmt.Sprintf("Can not connect to proxy : %s", err))
+			return
+		}
+	}
```
Comment (ineiti) [inline lib/monitor/proxy_test.go]: Same as with monitor.Listen - needed or not?

File: lib/monitor/proxy_test.go (hunk 2 above)
Comment (Liamsi) [inline lib/monitor/proxy_test.go]: This will be part of #163

File: lib/monitor/proxy_test.go (hunk 3 above)
Comment (Liamsi) [inline lib/monitor/proxy_test.go]: Also part of #163

**Comment by Liamsi** [comment]
Ready for merging

**Comment by Liamsi** [comment]
BTW: I think the error is related to this: https://github.com/golang/go/issues/11872#issuecomment-126183460
because the first test closes the connection but we still get a `getsockopt: connection refused`

**Comment by nikkolasg** [comment]
Since there will be quite some refactoring in #163 , for me it is OK to merge.

**Pull request merged**